source/mechanics/mechanics_utilities.cc

```diff
@@ -362,10 +383,23 @@ namespace fdl
                       }
                   }
 
+                if (touched_stress)
+                  {
+                    if (cell->material_id() != current_id)
+                      current_as = as_map[cell->material_id()];
+                    if (current_as)
+                      {
+                        auto view =
+                          make_array_view(accumulated_stresses.begin(),
+                                          accumulated_stresses.end());
+                        current_as->apply(cell, exemplar_quadrature, view);
+                      }
+                  }
+
```

Comment:
@marshallrdavey Mind taking a quick look here and at `ActiveStrain.h`? I think this is a reasonable way to implement what we want.